cmd/analyze.go

```diff
@@ -434,71 +435,125 @@ func (c *coremarkAnalyzer) Close() (err error) {
+	testMode     string
+	diskType     string
+	machineType  string
+	clientRegion string
+	serverRegion string
+
+	// Latency results.
+	latTestDuration   string
+	minLatMicrosec    string
+	meanLatMicrosec   string
+	latMicrosec_90    string
+	latMicrosec_99    string
+	maxLatMicrosec    string
+	latStdDevMicrosec string
+	txnRate           string
+
+	// Throughput results.
+	expectedThroughput string
+	numStreams         string
+	throughputDuration string
+	throughtputRes     string
+	throughputUnit     string
+	minThroughput      string
+	maxThroughput      string
+	meanThroughput     string
+	dateTime           string
+	recvBufferSize     string
+	sendBufferSize     string
+	timeSeriesPlotPath string
+
+	modtime time.Time
 }
 
 type netAnalyzer struct {
 	machineResults map[string]*networkResult
 	cloud          string
+	testMode       string
+}
+
+func newIntraAzNetAnalyzer(cloud string) resultsAnalyzer {
+	return &netAnalyzer{
+		cloud:          cloud,
+		machineResults: make(map[string]*networkResult),
+		testMode:       "intra-az",
+	}
 }
 
-func newNetAnalyzer(cloud string) resultsAnalyzer {
+func newCrossRegionNetAnalyzer(cloud string) resultsAnalyzer {
 	return &netAnalyzer{
 		cloud:          cloud,
 		machineResults: make(map[string]*networkResult),
+		testMode:       "cross-region",
 	}
 }
 
 func (n *netAnalyzer) Close() (err error) {
-	f, err := os.OpenFile(ResultsFile("net.csv", n.cloud), os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644)
+	fileName := fmt.Sprintf("%s-net.csv", n.testMode)
+	f, err := os.OpenFile(ResultsFile(fileName, n.cloud), os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644)
 	if err != nil {
 		return err
 	}
 	defer func() { err = f.Close() }()
 
-	fmt.Fprintf(f, "%s\n", netCSVHeader)
+	if _, err := fmt.Fprintf(f, "%s\n", netCSVHeader); err != nil {
+		return fmt.Errorf("cannot write network header to csv: %v", err)
+	}
+
 	for machineType, res := range n.machineResults {
 		fields := []string{
+			res.testMode,
 			n.cloud,
-			res.modtime.String(),
+			res.dateTime,
 			machineType,
-			fmt.Sprintf("%f", res.throughput),
-			res.throughputUnits,
-			fmt.Sprintf("%f", res.minLat),
-			fmt.Sprintf("%f", res.meanLat),
-			fmt.Sprintf("%f", res.p90Lat),
-			fmt.Sprintf("%f", res.p99Lat),
-			fmt.Sprintf("%f", res.maxLat),
-			fmt.Sprintf("%f", res.latStdDev),
-			fmt.Sprintf("%f", res.txnRate),
+			res.diskType,
+			res.clientRegion,
+			res.serverRegion,
+			res.minThroughput,
+			res.meanThroughput,
+			res.maxThroughput,
+			res.throughputUnit,
+			res.expectedThroughput,
+			res.numStreams,
+			res.recvBufferSize,
+			res.sendBufferSize,
+			res.throughputDuration,
+			res.latTestDuration,
+			res.minLatMicrosec,
+			res.meanLatMicrosec,
+			res.latMicrosec_90,
+			res.latMicrosec_99,
+			res.maxLatMicrosec,
+			res.latStdDevMicrosec,
+			res.txnRate,
+			res.timeSeriesPlotPath,
+		}
+		if _, err := fmt.Fprintf(f, "%s\n", strings.Join(fields, ",")); err != nil {
+			return fmt.Errorf("cannot output fields to the csv: %v", err)
```

Comment:
Nit: could include the file name in the error message to make it more descriptive.

Reply:
Done.